adb_graphics/figures/skewt.py

```diff
@@ -53,6 +54,45 @@ def __init__(self, ds, loc, **kwargs):
 
         self.max_plev = kwargs.get('max_plev', 0)
 
+    def _add_hydrometeors(self, hydro_subplot):
+
+        profiles = self.atmo_profiles # dictionary
+        pres = profiles.get('pres').get('data')
+        clwmr = profiles.get('clwmr').get('data') * 1000.
+        grle = profiles.get('grle').get('data') * 1000.
+        icmr = profiles.get('icmr').get('data') * 1000.
+        rwmr = profiles.get('rwmr').get('data') * 1000.
+        snmr = profiles.get('snmr').get('data') * 1000.
+
+        # set upper and lower limits for plot
+        clwmr = np.where((clwmr > 0.) & (clwmr < 1.e-4), 1.e-4, clwmr)
+        grle = np.where((grle > 0.) & (grle < 1.e-4), 1.e-4, grle)
+        icmr = np.where((icmr > 0.) & (icmr < 1.e-4), 1.e-4, icmr)
+        rwmr = np.where((rwmr > 0.) & (rwmr < 1.e-4), 1.e-4, rwmr)
+        snmr = np.where((snmr > 0.) & (snmr < 1.e-4), 1.e-4, snmr)
+
+        clwmr = np.where((clwmr > 10.), 10., clwmr)
+        grle = np.where((grle > 10.), 10., grle)
+        icmr = np.where((icmr > 10.), 10., icmr)
+        rwmr = np.where((rwmr > 10.), 10., rwmr)
+        snmr = np.where((snmr > 10.), 10., snmr)
+
+        # Pressure vs mixing ratios
+        hydro_subplot.plot(clwmr, pres, 'blue', marker="s", markersize=6, fillstyle="none")
+        hydro_subplot.plot(grle, pres, 'orange', marker="D", markersize=6, fillstyle="none")
+        hydro_subplot.plot(icmr, pres, 'red', marker="^", markersize=6, fillstyle="none")
+        hydro_subplot.plot(rwmr, pres, 'cyan', marker="o", markersize=6, fillstyle="none")
+        hydro_subplot.plot(snmr, pres, 'purple', marker="*", markersize=6, fillstyle="none")
+
+        # hydrometeor legend
+        handles = []
+        handles.append(mpatches.Patch(facecolor='none', edgecolor="purple", label="SNOW"))
+        handles.append(mpatches.Patch(facecolor='none', edgecolor="cyan", label="RAIN"))
+        handles.append(mpatches.Patch(facecolor='none', edgecolor="blue", label="CWAT"))
+        handles.append(mpatches.Patch(facecolor='none', edgecolor="orange", label="GRPL"))
+        handles.append(mpatches.Patch(facecolor='none', edgecolor="red", label="CICE"))
+        plt.legend(handles=handles, loc=[4.25, -0.8])
```

Comment:
I think this section could benefit from a loop over variable settings in a dictionary:

```
mixing_ratios = {
    'clwmr': {'color': 'blue', 'label': 'CWAT", 'marker': 's'},
   ...}
   
for mixr, settings in mixing_ratios.items():
    profile = profiles.get(mixr).get('data') * 1000.
    profile = np.where((profile > 0.) & (profile < 1.e-4), 1.e-4, profile)
    profile = np.where((profile > 10.), 10., profile)

    hydro_subplot.plot(profile, pres, settings.get('color'), marker=setting.get('marker'), markersize=6, fillstyle="none")

    handles.append(mpatches.Patch(facecolor='none', edgecolor=settings.get('color'), label=settings.get('label')

plt.legend(handles....)

```

Reply:
thanks, I put these changes in.  much cleaner.